ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouterMetrics.java

```diff
@@ -930,25 +954,25 @@ public void close() {
     @Override
     public void run() {
       double quantile = routerConfig.routerLatencyToleranceQuantile;
-      for (Map.Entry<Resource, Histogram> resourceToHistogram : getBlobLocalDcResourceToLatency.entrySet()) {
+      for (Map.Entry<Resource, CachedHistogram> resourceToHistogram : getBlobLocalDcResourceToLatency.entrySet()) {
         Resource resource = resourceToHistogram.getKey();
         Histogram histogram = resourceToHistogram.getValue();
         logger.debug("{} GetBlob local DC latency histogram {}th percentile in ms: {}", resource.toString(),
             quantile * 100, histogram.getSnapshot().getValue(quantile));
```

Comment:
Also suggest changing `histogram.getSnapshot().getValue(quantile)` to `histogram.getCachedValue()` in this method.